lib/EarlyFunctionAnalysis/EarlyFunctionAnalysis.cpp

```diff
@@ -843,6 +877,98 @@ void FunctionEntrypointAnalyzer::interproceduralPropagation() {
   }
 }
 
+std::optional<abi::RegisterState::Values>
+FEA::tryGetRegState(model::Register::Values RegisterValue,
+                    const ABIAnalyses::RegisterStateMap &ABIRegisterMap) {
+  llvm::StringRef Name = model::Register::getCSVName(RegisterValue);
+  if (llvm::GlobalVariable *CSV = M.getGlobalVariable(Name, true)) {
+    auto It = ABIRegisterMap.find(CSV);
+    if (It != ABIRegisterMap.end())
+      if (abi::RegisterState::isYesOrDead(It->second))
+        return It->second;
+  }
+
+  return std::nullopt;
+}
+
+void FEA::initializeMapForDeductions(FunctionSummary &Summary,
+                                     abi::RegisterState::Map &Map) {
+  using namespace abi;
+
+  auto GPRArguments = listGeneralPurposeArgumentRegisters(Binary->DefaultABI);
+  auto GPRRetVals = listGeneralPurposeReturnValueRegisters(Binary->DefaultABI);
+  auto VRArguments = listVectorArgumentRegisters(Binary->DefaultABI);
+  auto VRRetVals = listVectorReturnValueRegisters(Binary->DefaultABI);
+
+  const auto &Results = Summary.ABIResults;
+  for (auto &GPRArg : GPRArguments)
+    if (auto MaybeState = tryGetRegState(GPRArg, Results.ArgumentsRegisters))
```

Comment:
I'm confused:
* Why do you initialize them only if `isYesOrDead`?
* Why do you pass only `listGeneralPurposeArgumentRegisters` and not all of them? I don't think EFA should be aware of any detail of ABI deduction such as  `listGeneralPurposeArgumentRegisters`. It should just know about `applyRegisterStateDeductions`.